cmp/filter/impl/QueryEngine.js

```diff
@@ -66,7 +67,15 @@ export class QueryEngine {
     }
 
     //------------------------------------------------------------------------
-    // 1) No op yet, so field not fixed -- get field or value matches.
+    // 1) No query -- return all field suggestions
+    //------------------------------------------------------------------------
+    suggestAllFields() {
+        // Suggest all for the user to select on their way to a more targeted query.
```

Comment:
for clarity/robustness think we could call this method `whenNoQuery` `noQueryPresent` etc.  and then within we can say return suggestFieldsWhenEmpty ? this.getFieldOpts() : []